specs/online_order_spec.rb

```diff
@@ -1,90 +1,148 @@
-# require_relative '../lib/online_order'
+require_relative '../lib/online_order'
 # You may also need to require other classes here
 
 # Because an OnlineOrder is a kind of Order, and we've
 # already tested a bunch of functionality on Order,
 # we effectively get all that testing for free! Here we'll
 # only test things that are different.
 
-xdescribe "OnlineOrder" do
+describe "OnlineOrder" do
   describe "#initialize" do
     it "Is a kind of Order" do
-      # Check that an OnlineOrder is in fact a kind of Order
-
-      # Instatiate your OnlineOrder here
-      # online_order =
-      # online_order.must_be_kind_of Grocery::Order
+      id = 1
+      products = {"Lobster" => 17.18, "Annatto seed" => 58.38, "Camomile" => 83.21}
+      customer = 25
+      status = :complete
+
+      online_order = Grocery::OnlineOrder.new(id, products, customer, status)
+      online_order.must_respond_to :products
+      online_order.products.length.must_equal 3
+      online_order.must_be_kind_of Grocery::Order
     end
 
     it "Can access Customer object" do
-      # TODO: Your test code here!
+      id = 1
+      products = {"Lobster" => 17.18, "Annatto seed" => 58.38, "Camomile" => 83.21}
+      customer = 25
+      status = :complete
+
+      online_order = Grocery::OnlineOrder.new(id, products, customer, status)
+
+      online_order.must_respond_to :id
+      online_order.id.must_equal id
     end
 
     it "Can access the online order status" do
-      # TODO: Your test code here!
+      id = 1
+      products = {"Lobster" => 17.18, "Annatto seed" => 58.38, "Camomile" => 83.21}
+      customer = 25
+      status = :complete
+
+      online_order = Grocery::OnlineOrder.new(id, products, customer, status)
+
+      online_order.must_respond_to :status
+      online_order.status.must_equal :complete
     end
   end
 
   describe "#total" do
     it "Adds a shipping fee" do
-      # TODO: Your test code here!
+      id = 1
+      products = {"Lobster" => 17.18, "Annatto seed" => 58.38, "Camomile" => 83.21}
+      customer = 25
+      status = :complete
+
+      online_order = Grocery::OnlineOrder.new(id, products, customer, status)
+
+      online_order.total.must_equal 180.68
     end
 
     it "Doesn't add a shipping fee if there are no products" do
-      # TODO: Your test code here!
+      id = 1
+      products = {}
+      customer = 25
+      status = :complete
+
+      online_order = Grocery::OnlineOrder.new(id, products, customer, status)
+
+      online_order.total.must_equal nil
     end
   end
 
   describe "#add_product" do
     it "Does not permit action for processing, shipped or completed statuses" do
-      # TODO: Your test code here!
+      id = 1
+      products = {"Lobster" => 17.18, "Annatto seed" => 58.38, "Camomile" => 83.21}
+      customer = 25
+      status = :complete
+
+      online_order = Grocery::OnlineOrder.new(id, products, customer, status)
+
+      online_order.add_product("Annatto seed", 58.38).must_be_nil
```

Comment:
Like `Order` this should return `true` if the product was added and false otherwise.